layers.go

```diff
@@ -527,7 +527,7 @@ func (r *layerStore) SetMetadata(id, metadata string) error {
 }
 
 func (r *layerStore) tspath(id string) string {
-	return filepath.Join(r.layerdir, id+tarSplitSuffix)
+	return filepath.Join(DefaultStoreOptions.GraphRoot, DefaultStoreOptions.GraphDriverName+"-layers", id+tarSplitSuffix)
 }
```

Comment:
If you setup a layer/image on an nfs share it uses the path associated with the image to create the tarsplit.  It passes in /var/lib/nfs/storage which I mount as readonly.